gradle/libs.versions.toml

```diff
@@ -1,4 +1,6 @@
 [versions]
+java-compilation = "20" # `actions/setup-java` should install the same version
+java-target = "8"
```

Comment:
This is my new proposal to define project's Java compatibility. I hope the names are clear 😅 

`java-target` is much more important as it will affect consumers, it defines bytecode compatibility the produced `.jar`s will comply to.

Reply:
Maybe change to something like:
```
# The java-compilation version is the latest supported (not necessarily LTS) version of Java. It should be identical to Java-version used in `actions/setup-java`
java-compilation = "20"
# The java-target version is the lowest supported LTS version of Java. Jar's produced are bytecode compatible with this version.
java-target = "8"
```